### ISSUE TITLE
Integration with Swagger 2.0

### ISSUE THREAD

**Issue by johnament** [hammock-project/hammock#119]
Add support for Swagger 2.0 https://github.com/swagger-api/swagger-core/tree/2.0

- Single coordinate to depend on that adds Swagger 2.0 APIs.
- It should add swagger support any JAX-RS runtime.

**Comment by derekm** [comment]
The benefit of pushing the integration into Swagger 2.0's jars would be allowing the user to control the Swagger version upgrades (being able to upgrade Swagger without relying on Hammock releases).

**Comment by johnament** [comment]
@derekm for any dependency that hammock brings in, you can mark the dependencies as excluded, or if you declare the dependencies before the hammock JAR maven will use those instead of the ones hammock brings in.

**Comment by derekm** [comment]
Ok, I didn't know about the dependency order trick! It's less explicit, but maybe it beats the exclude approach. Thanks for the tip!

**Comment by derekm** [comment]
@johnament -- RestAssured looks very interesting! ... Btw, I've got a highly generic set of CrudResource interfaces and abstract classes + CrudEndpoint/CrudService + JPA & JAXB type transformers. Jersey Server has no problems presenting extensions of these CRUD services for endpoints of random types, but Jersey Proxy Client struggles when it hits CrudResource<T> off of some concrete `interface WhateverResources extends CrudResource<Whatever>`. This CRUD logic refactor saves a ton of code. Do you think MP Rest Client 1.0 would be able to handle it? I'll have to send my CRUD refactor your way!

**Comment by derekm** [comment]
Base & Interface:
```
+/**
+ * A generic CRUD Endpoint implementation.
+ *
+ * @author dmoore
+ *
+ * @param <T> return type & service transform return type
+ * @param <S> service type
+ * @param <E> service entity type
+ * @param <X> service transformation type
+ * @param <I> input/output transformation type
+ */
+public abstract class CrudEndpoint<T,
+                                   S extends CrudService<E, X, T, I>,
+                                   E,
+                                   X extends Function<E, T>,
+                                   I extends EntityTransformer<T, E>
+                                   >
+        implements CrudResource<T> {
```
Implementation:
```
+public class AnswersEndpoint
+        extends CrudEndpoint<QuestionAnswer,
+                             AnswerService,
+                             Answer,
+                             AnswerTransformer,
+                             QuestionAnswerTransformer>
+        implements AnswersResource {
```
:D

**Comment by johnament** [comment]
What do the annotations look like on the actual interfaces?

**Comment by derekm** [comment]
```
+public interface CrudResource<T> {
+
+    @GET
+    @Consumes({
+            MediaType.APPLICATION_JSON,
+            MediaType.APPLICATION_XML,
+            })
+    @Produces({
+            MediaType.APPLICATION_JSON,
+            MediaType.APPLICATION_XML,
+            })
+    public List<T> get();
+
+    @POST
+    @Consumes({
+            MediaType.APPLICATION_JSON,
+            MediaType.APPLICATION_XML,
+            })
+    @Produces({
+            MediaType.APPLICATION_JSON,
+            MediaType.APPLICATION_XML,
+            })
+    public T post(T answer);
+
+}
```
and
```
public interface AnswersResource extends CrudResource<QuestionAnswer> {}
```

**Comment by derekm** [comment]
`swagger-jaxrs2` is now CDI-enabled in `swagger-core` 2.0.0-SNAPSHOT, presumably to be released in 2.0.0-rc4: https://github.com/swagger-api/swagger-core/pull/2627
